joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -408,16 +357,81 @@ JointTrajectoryController::on_configure(const rclcpp_lifecycle::State & previous
     std::bind(&JointTrajectoryController::feedback_setup_callback, this, _1)
   );
 
-  set_op_mode(hardware_interface::OperationMode::INACTIVE);
-
   return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
 }
 
+// Fill ordered_interfaces with references to the matching interfaces
+// in the same order as in joint_names
+template<typename T>
+bool get_ordered_interfaces(
+  std::vector<T> & unordered_interfaces, const std::vector<std::string> & joint_names,
+  const std::string & interface_type, std::vector<std::reference_wrapper<T>> & ordered_interfaces)
+{
+  for (const auto & joint_name : joint_names) {
+    for (auto & command_interface : unordered_interfaces) {
+      if ((command_interface.get_name() == joint_name) &&
+        (command_interface.get_interface_name() == interface_type))
+      {
+        ordered_interfaces.push_back(std::ref(command_interface));
```

Comment:
Is it necessary to use a reference wrapper?

Reply:
I didn't find a better alternative, can't copy it for obvious reasons, can use a raw reference because the elements of a vector need to be default constructible and didn't like a raw pointer because it opens up the question as to who should delete it.

Reply:
I tried to use raw references too but the compiler told me off as there's a move-constructor defined